init.go

```diff
@@ -0,0 +1,130 @@
+package surge
+
+// This file implements the centralized initialization framework for surge.
+// The core interface is Initializer. Each file or module that needs to have some
+// initialization logic before calling main should create a structure which includes
+// BaseInitializer structure. The structure should redefine the Initialize() method
+// which should call the function that performs initialization. Each initialization
+// can have a dependency list of Initializers. The dependency should be met before
+// the initializer is called.
+// One example: config should be called before model. So model initialization
+// will have a dependency set for config.
+//
+// With this new framework, no files other than this file will have the init method.
+// All files or modules that need initialization will define their own initializer and
+// set the dependency correctly and then add it to the global initializers list using
+// the AddInitializer() function. This should be called as a variable initialization in
+// each file so that they get called befor the init() function here.
+
+import (
+	"fmt"
+	)
+
+
+type InitializerState int
+
+const (
+	InitializerNotProcessed InitializerState = iota
+	InitializerBlocked
+	InitializerDeadlocked
+	InitializerProcessed
```

Comment:
Better, assert on a deadlock, and then fix the code. No need to have a retcode for this